databricks/koalas/frame.py

```diff
@@ -2836,6 +2836,71 @@ def merge(left, right: 'DataFrame', how: str = 'inner',
 
         return DataFrame(joined_table.select(*exprs))
 
+    def append(self, other: 'DataFrame', ignore_index: bool = False,
+               verify_integrity: bool = False, sort: bool = False) -> 'DataFrame':
+        """
+        Append rows of other to the end of caller, returning a new object.
+
+        Columns in other that are not in the caller are added as new columns.
+
+        Parameters
+        ----------
+        other : DataFrame or Series/dict-like object, or list of these
+            The data to append.
+
+        ignore_index : boolean, default False
+            If True, do not use the index labels.
+
+        verify_integrity : boolean, default False
+            If True, raise ValueError on creating index with duplicates.
+
+        sort : boolean, default False
+            Currently not supported.
+
+        Returns
+        -------
+        appended : DataFrame
+
+        Examples
+        --------
+        >>> df = ks.DataFrame([[1, 2], [3, 4]], columns=list('AB'))
+
+        >>> df.append(df)
+           A  B
+        0  1  2
+        1  3  4
+        0  1  2
+        1  3  4
+
+        >>> df.append(df, ignore_index=True)
+           A  B
+        0  1  2
+        1  3  4
+        2  1  2
+        3  3  4
+        """
+        if isinstance(other, ks.Series):
+            raise ValueError("DataFrames.append() does not support appending Series to DataFrames")
+        if sort:
+            raise ValueError("The 'sort' parameter is currently not supported")
+
+        index_columns = self._metadata.index_columns
+        if len(index_columns) != len(other._metadata.index_columns):
+            raise ValueError("Both DataFrames have to have the same number of index levels")
```

Comment:
Can we skip this check if `ignore_index is True`?

Reply:
No, because (right now) Koalas cannot concat DataFrames with different index levels like in pandas:

```python
>>> df1 = pd.DataFrame({'A':[1,2,3]},index=[1,2,3])
>>> df2 = pd.DataFrame({'A':[1,2,3]},index=[[1,2,3],[4,5,6]])
>>> df1.append(df2)
        A
1       1
2       2
3       3
(1, 4)  1
(2, 5)  2
(3, 6)  3
```

Koalas' implementation of `concat()` will throw an exception for this case. While we could fix `concat()` to correctly handle this, I believe that this is really an uncommon use case so I'd rather not waste too much energy on this.

Reply:
I mean `ignore_index is True` .. :

```py
>>> import pandas as pd
>>> df1 = pd.DataFrame({'A':[1,2,3]},index=[1,2,3])
>>> df2 = pd.DataFrame({'A':[1,2,3]},index=[[1,2,3],[4,5,6]])
>>> df1.append(df2, ignore_index=True)
   A
0  1
1  2
2  3
3  1
4  2
5  3
```

`ks.concat` supports such case:

```py
>>> import databricks.koalas as ks
>>> kdf1 = ks.DataFrame({'A':[1,2,3]},index=[1,2,3])
>>> kdf2 = ks.DataFrame({'A':[1,2,3]},index=[[1,2,3],[4,5,6]])
>>> ks.concat([kdf1, kdf2], ignore_index=True)
   A
0  1
1  2
2  3
3  1
4  2
5  3
```